packages/headless/src/features/search/search-slice.ts

```diff
@@ -53,5 +57,8 @@ export const searchReducer = createReducer(
     });
     builder.addCase(executeSearch.pending, handlePendingSearch);
     builder.addCase(fetchMoreResults.pending, handlePendingSearch);
+    builder.addCase(clearSearchReponse, (state) => {
+      state.response = getSearchInitialState().response;
+    });
```

Comment:
What do you think of ?
```suggestion
    builder.addCase(clearSearchReponse, (state) => ({
      ...state,
      response = getSearchInitialState().response,
    }));
```

Reply:
The pattern in the rest of the reducers here is to do an assignment rather than return the state so I would opt to leave it as is.

Reply:
It uses immer, so you don't need to https://redux-toolkit.js.org/usage/immer-reducers

Reply:
Ok so the toolkit added a patch for those who still want to use unsafe state mutation.